x-pack/plugins/security_solution/public/overview/components/sidebar/sidebar.test.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import { waitFor } from '@testing-library/react';
+import { TestProviders } from '../../../common/mock';
+import { Sidebar } from './sidebar';
+import { useGetUserCasesPermissions, useKibana } from '../../../common/lib/kibana';
+import { casesPluginMock } from '../../../../../cases/public/mocks';
+import { CasesUiStart } from '../../../../../cases/public';
+
+jest.mock('../../../common/lib/kibana');
+
+const useKibanaMock = useKibana as jest.MockedFunction<typeof useKibana>;
+
+describe('Sidebar', () => {
+  let casesMock: jest.Mocked<CasesUiStart>;
+
+  beforeEach(() => {
+    casesMock = casesPluginMock.createStartContract();
+    casesMock.getRecentCases.mockImplementation(() => <>{'test'}</>);
+    useKibanaMock.mockReturnValue(({
+      services: {
+        cases: casesMock,
+        application: {
+          // these are needed by the RecentCases component if it is rendered.
+          navigateToApp: jest.fn(),
+          getUrlForApp: jest.fn(() => ''),
+        },
+      },
+    } as unknown) as ReturnType<typeof useKibana>);
+  });
+
+  it('does not render the recently created cases section when the user does not have read permissions', async () => {
+    (useGetUserCasesPermissions as jest.Mock).mockReturnValue({
+      crud: false,
+      read: false,
+    });
+
+    await waitFor(() =>
+      mount(
+        <TestProviders>
+          <Sidebar recentTimelinesFilterBy={'favorites'} setRecentTimelinesFilterBy={() => {}} />
+        </TestProviders>
+      )
+    );
+
+    expect(casesMock.getRecentCases).not.toHaveBeenCalled();
```

Comment:
Shouldn't we check if the component is does not exists?

Reply:
I might be misunderstanding you but I think if we wanted to check that the component exists we'd have to grab the data test subj from the cases plugin and make sure it doesn't exist. I feel like that is relying on implementation details of how the cases plugin implements the recent cases. Doing it like this we ensure that we don't call the cases plugin at all and therefore shouldn't render anything.

I guess another option would be to create an empty wrapper component that defines a data test subj and ensure it doesn't get rendered.